src/state/opamEnv.ml

```diff
@@ -498,45 +498,101 @@ let env_hook_script shell =
       ^ script)
     (env_hook_script_base shell)
 
-let source root shell f =
+let if_expr ?(level=0) ~icond ~ithen ?ielse shell =
+  let pad = String.concat "" (List.init (level*2) (fun _ -> " ")) in
+  let ielse_expr else_opt = match else_opt with
+    | None -> ""
+    | Some e -> Printf.sprintf "else\n  %s%s\n" pad e
+  in
+  let ithen = pad ^ ithen ^ "\n" in
+  match shell with
+  | SH_sh | SH_bash ->
+    Printf.sprintf "%sif [ %s ]; then\n  %s%s%sfi"
+      pad icond ithen (ielse_expr ielse) pad
+  | SH_zsh ->
+    Printf.sprintf "%sif [[ %s ]]; then\n  %s%s%sfi"
+      pad icond ithen (ielse_expr ielse) pad
```

Comment:
Why is the bash / zsh distinction needed here ? I thought both syntaxes existed in both, with `[[` allowing more specific built-ins. `[` should behave the same for zsh as for bash.

Reply:
Simply because I took if expression from **[`if_interactive_script`](https://github.com/ocaml/opam/blob/3b729c9e770a083d204e710c62ba005d2b14b69a/src/state/opamEnv.ml#L520), and it set this sh/zsh difference